### ISSUE TITLE
/bin/sh not able to parse bash functions

### ISSUE THREAD

**Issue by miketheman** [Glavin001/atom-beautify#234]
Command failed: /bin/sh -c [ -f ~/.bash_profile ] && source ~/.bash_profile;[ -f ~/.bashrc ] && source ~/.bashrc;rbeautify "/var/folders/78/3jyqdny925n348fpr6wh23bc0000gn/T/input115211-94410-1yp2t0n" /Users/miketheman/.bash_it/lib/helpers.bash: line 76: `bash-it': not a valid identifier

This error occurs since I have some functions being loaded in my bash_profile that to not work with the default `/bin/sh` shell on OSX.

This is due to the default behavior of child_process and exec. See more here: https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback

I tried tinkering with the code to allow accepting `bash` as the new shell, since you end up sourcing the functions declared in bash_profle anyhow, but I couldn't get it working.

**Comment by lazee** [comment]
Kind of the same problem here since I'm using zshell and have not configured the correct environment in Bash config.

**Comment by Glavin001** [comment]
This will be consequently fixed by #164 
For now, if you are able to comment out https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/cli-beautify.coffee#L49-L50 in your local installations.

Has anyone tried passing `bash` as the `shell` to https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/cli-beautify.coffee#L57 ? I am unable to reproduce and also in the middle of exams, so it'll be a few weeks before I have time to look into this further.

**Comment by miketheman** [comment]
I've been able to make this kind of thing work on master code:

```
...
                if typeof cmd is "string"

                  config =
                    env: process.env
                    shell: 'bash'

                  isWin = /^win/.test(process.platform)
...
```

This might work for `zshell` as well.

I know you're working on something better longer term, so I didn't know if it was worthwhile sending a PR for this change, or leaving it for now, since there's a workaround.

I guess it would most make sense to try and detect the user's `$SHELL` variable and react to that, however that brings into problems of detecting where the Atom instance was started from - it you launch via Spotlight, Alfred or Start Menu, you won't have access to the envvars during launch time.

See [here](https://github.com/ioquatix/script-runner/blob/599264e8185cab063e88c7ac1f983d15fbe044e7/lib/script-runner.coffee#L34-L64) for how another package 'solved' this problem.

**Comment by Glavin001** [comment]
This should not longer be applicable now that #282 is published. Published to v0.25.0 and above.
